core/src/Kokkos_Macros.hpp

```diff
@@ -535,8 +535,10 @@
 
 #if defined(KOKKOS_ENABLE_CXX17) || defined(KOKKOS_ENABLE_CXX20)
 #define KOKKOS_ATTRIBUTE_NODISCARD [[nodiscard]]
+#define KOKKOS_CONSTEXPR_17 constexpr
```

Comment:
Hmmm... So far we have refrained from introducing `if constexpr` workarounds. Is this anywhere required in this pull request?

Reply:
Not a requirement, but it can potentially skip a kernel instantiation if the user supplies a launch bound, e.g., [here](https://github.com/kokkos/kokkos/pull/3953/files#diff-078df6e850082d467d9c91f0888ce3b7b5f5f0fcde51bc91a00845ec6dd9bfb0R334)

Happy enough to remove it.

Reply:
If you could, remove it and add a note to make it if constexpr when we require 17

Reply:
Sounds good

Reply:
Fixed in latest.